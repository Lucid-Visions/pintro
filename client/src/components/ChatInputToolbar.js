import React from 'react';
import { StyleSheet, View, TextInput, TouchableOpacity, Image } from 'react-native';

/**
 *
 */
export default class ChatInputToolbar extends React.Component {
  constructor(props) {
      super(props);

      this.state = {
        message: ""
      }

      this.messageInput = React.createRef();

      this.onSendHandler = this.onSendHandler.bind(this);
      this.onChangeTextHandler = this.onChangeTextHandler.bind(this);
  };

  onSendHandler(){
    if(this.state.message !== ""){
      this.props.onSend(this.state.message);

      // clear the input
      this.setState({ message: "" });
      this.messageInput.current.clear();
    }
  }

  onChangeTextHandler(newValue){
    this.setState({ message: newValue });
  }

  render() {
    return(
      <View style={styles.container}>
        <View style={styles.innerContainer}>

          <TextInput
              multiline={true}
              style={{ flex: 1, marginLeft: 25, marginRight: 10 }}
              onChangeText={this.onChangeTextHandler}
              ref={this.messageInput}
              placeholder={"Your message..."}
          />

          <TouchableOpacity onPress={this.onSendHandler} style={styles.sendButton}>
            <Image source={require('../assets/rightArrowBlack.png')} resizeMode={'center'} style={{height: 42, width: 42}}/>
          </TouchableOpacity>

        </View>
      </View>
    );
  }
};

const styles = StyleSheet.create({
  container: {
    maxHeight: 120,
    paddingTop: 10,
    paddingBottom: 10,
    borderColor: 'lightgray',
    borderTopWidth: 1.0
  },
  innerContainer: {
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    paddingRight: 5,
    paddingTop: 5,
    paddingBottom: 5,
    marginRight: 30,
    marginLeft: 30,
    borderColor: 'gray',
    borderWidth: 1.5,
    borderRadius: 30
  },
  sendButton: {
    alignSelf: "flex-end",
    backgroundColor: "orange",
    borderRadius: 180
  }
});
