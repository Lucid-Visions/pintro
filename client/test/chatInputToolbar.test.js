import React from 'react';
import ReactDOM from 'react-dom';
import renderer, { act } from 'react-test-renderer';
import ChatInputToolbar from '../src/components/ChatInputToolbar';
import { TextInput } from 'react-native';

test('Chat input toolbar renders correctly', () => {
    const component = renderer.create(<ChatInputToolbar />).toJSON();
    expect(component).toMatchSnapshot();
});

/**test('State changes when input value changes', async () => {
    const component = renderer.create(<ChatInputToolbar />);
    const input = component.root.findByType(TextInput);
    const mockChange = {
        target: {
            value: 'test'
        }
    }
    act(() => {
         input.dispatchEvent(new TextEvent('change', mockChange));
    })
     expect(component.state.message).toBe('test');
})
*/