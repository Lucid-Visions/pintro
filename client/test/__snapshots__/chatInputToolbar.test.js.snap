// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat input toolbar renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "lightgray",
      "borderTopWidth": 1,
      "maxHeight": 120,
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "gray",
        "borderRadius": 30,
        "borderWidth": 1.5,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginLeft": 30,
        "marginRight": 30,
        "paddingBottom": 5,
        "paddingRight": 5,
        "paddingTop": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="Your message..."
      rejectResponderTermination={true}
      style={
        Object {
          "flex": 1,
          "marginLeft": 25,
          "marginRight": 10,
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "orange",
          "borderRadius": 180,
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="center"
        source={1}
        style={
          Object {
            "height": 42,
            "width": 42,
          }
        }
      />
    </View>
  </View>
</View>
`;
